rust/rsc/src/database.rs

```diff
@@ -193,19 +192,92 @@ pub async fn read_dbonly_blob_store<T: ConnectionTrait>(
 
 // ----------             Read             ----------
 pub async fn read_job<T: ConnectionTrait>(db: &T, id: Uuid) -> Result<Option<job::Model>, DbErr> {
-    job::Entity::find_by_id(id).one(db).await
+    Job::find_by_id(id).one(db).await
 }
 
 pub async fn search_jobs_by_label<T: ConnectionTrait>(
     db: &T,
     label: &String,
 ) -> Result<Vec<job::Model>, DbErr> {
-    job::Entity::find()
+    Job::find()
         .filter(job::Column::Label.like(label))
         .order_by_asc(job::Column::Label)
         .all(db)
         .await
 }
+
+pub async fn count_jobs<T: ConnectionTrait>(db: &T) -> Result<u64, DbErr> {
+    Job::find().count(db).await
+}
+
+#[derive(Debug, FromQueryResult)]
+pub struct TimeSaved {
+    pub savings: i64,
+}
+
+pub async fn time_saved<T: ConnectionTrait>(db: &T) -> Result<Option<TimeSaved>, DbErr> {
+    TimeSaved::find_by_statement(Statement::from_string(
+        DbBackend::Postgres,
+        r#"
+        SELECT CAST(round(sum(savings)) as INT8) as savings
+        FROM (
+            SELECT h.hits * j.runtime as savings
```

Comment:
`hits` starts at 0 when the job is posted right? Don't want to have an off-by-one error here.

Reply:
yep! hits starts at 0 and is only incremented when a "cache hit" is returned from the cache lookup